src/TwoOutOfThreeColumnVisitor.cpp

```diff
@@ -0,0 +1,148 @@
+                                for(int j=0; j<3; ++j) {
+
+                                        // if the columns already have this value from another Region, pass
+                                        if(possibleColumns[j]) {
+                                                possibles.push_back(vector<bool>(3, true));
+                                                continue;
+                                        } else {
+                                                possibles.push_back(vector<bool>(3, false));
+                                        }
+                                        
+                                        // This is the same for all j
+                                        possibles[j][0] = ioGrid.getRegion(3*((i+0)%3) + (main_col+1)%3).topRow().isValuePresent(value) || ioGrid.getRegion(3*((i+0)%3) + (main_col+2)%3).topRow().isValuePresent(value); 
+                                        possibles[j][1] = ioGrid.getRegion(3*((i+0)%3) + (main_col+1)%3).middleRow().isValuePresent(value) || ioGrid.getRegion(3*((i+0)%3) + (main_col+2)%3).middleRow().isValuePresent(value); 
+                                        possibles[j][2] = ioGrid.getRegion(3*((i+0)%3) + (main_col+1)%3).bottomRow().isValuePresent(value) || ioGrid.getRegion(3*((i+0)%3) + (main_col+2)%3).bottomRow().isValuePresent(value); 
+                                        
+                                        // Cannot be place in cells that are not empty
+                                        if(j == 0) {
+                                                fillableTriplet = regions[0].leftColumn();
+                                        } else if (j == 1) {
+                                                fillableTriplet = regions[0].middleColumn();
+                                        } else if (j == 2) {
+                                                fillableTriplet = regions[0].rightColumn();
+                                        }
+
+                                        if(!fillableTriplet.getFirst()->isEmpty()) {
+                                                possibles[j][0] = true;
+                                        }
+                                        if(!fillableTriplet.getSecond()->isEmpty()) {
+                                                possibles[j][1] = true;
+                                        }
+                                        if(!fillableTriplet.getThird()->isEmpty()) {
+                                                possibles[j][2] = true;
+                                        }
+                                }
+
+                                // If there is only one possible position where it could be place it
+                                position = findOnlyOneNegativeInCols(possibles);
+                                if (position.first!=-1) {
+                                        if(position.first == 0) {
```

Comment:
Please simplify this using; https://github.com/albanD/sudoku-solver/blob/master/src/RegionHolder.cpp#L25